### DISCUSSION TITLE
Logging _middleware. Response.on("finish")

### DISCUSSION THREAD

**Discussion by maxtotheguenther** [vercel/next.js#32701] [Help]
Hey guys,

Currently im trying to measure the time nextjs needs for serving the response.
For this use case I tried to implement this logic inside a "global" /pages/_middleware.ts.

```ts
export function middleware(req: NextRequest, event: NextFetchEvent) {
  const start = performance.now();
  const response = NextResponse.next();

  // Something like
  response.on("finish", () => {
    const duration = performance.now() - start;
    // log it baby
  });
}
```
Unfortunately NextResponse is an abstraction of the actual Express Response :/
So I cannot add an event listener. Is there a way to implement such logic?
The logging example provided in the Middleware documentation is missing

**Comment by tomvanrossom** [comment]
Just use these imports and then you are fine:

import type { NextFetchEvent, NextRequest } from 'next/server'
import {NextResponse} from 'next/server'

**Reply by maxtotheguenther** [reply]
@tomvanrossom 
Im using those types. But NextResponse does not extend Express Request, Response.
So I cannot register a finish callback
